mscore/prefsdialog.cpp

```diff
@@ -250,7 +247,248 @@ PreferenceDialog::PreferenceDialog(QWidget* parent)
+                                                }),
+
+      };
+
+      uiRelatedWidgets = vector<PreferenceItem*>{
+                  new BoolPreferenceItem(PREF_UI_CANVAS_BG_USECOLOR, bgColorButton, nullptr, [&](){ updateBgView(preferences.getBool(PREF_UI_CANVAS_BG_USECOLOR)); }),
+                  new ColorPreferenceItem(PREF_UI_CANVAS_BG_COLOR, bgColorLabel, nullptr, [](){;}),
+                  new BoolPreferenceItem(PREF_UI_CANVAS_FG_USECOLOR, fgColorButton, nullptr, [&](){ updateFgView(preferences.getBool(PREF_UI_CANVAS_FG_USECOLOR)); }),
+                  new BoolPreferenceItem(PREF_UI_CANVAS_FG_USECOLOR_IN_PALETTES, fgUseColorInPalettes, nullptr, [](){;}),
+                  new ColorPreferenceItem(PREF_UI_CANVAS_FG_COLOR, fgColorLabel, nullptr, [](){;}),
+                  new StringPreferenceItem(PREF_UI_CANVAS_BG_WALLPAPER, bgWallpaper, nullptr, [](){;}),
+                  new StringPreferenceItem(PREF_UI_CANVAS_FG_WALLPAPER, fgWallpaper, nullptr, [](){;}),
+                  new BoolPreferenceItem(PREF_UI_CANVAS_MISC_ANTIALIASEDDRAWING, drawAntialiased),
+                  new BoolPreferenceItem(PREF_UI_CANVAS_SCROLL_LIMITSCROLLAREA, limitScrollArea),
+                  new IntPreferenceItem(PREF_UI_CANVAS_MISC_SELECTIONPROXIMITY, proximity),
+                  new IntPreferenceItem(PREF_UI_THEME_ICONWIDTH, iconWidth),
+                  new IntPreferenceItem(PREF_UI_THEME_ICONHEIGHT, iconHeight),
+                  new StringPreferenceItem(PREF_UI_THEME_FONTFAMILY, fontFamily, nullptr,
+                                          [&]() { // update function
+                                                auto currFontFamily = preferences.getString(PREF_UI_THEME_FONTFAMILY);
+                                                if (-1 == fontFamily->findText(currFontFamily))
+                                                      fontFamily->addItem(currFontFamily);
+                                                fontFamily->setCurrentIndex(fontFamily->findText(currFontFamily));
+                                                }),
+                  new IntPreferenceItem(PREF_UI_THEME_FONTSIZE, fontSize),
+                  new CustomPreferenceItem(PREF_UI_APP_GLOBALSTYLE, styleName,
+                                          [&]() { // apply function
+                                                preferences.setCustomPreference<MuseScoreStyleType>(PREF_UI_APP_GLOBALSTYLE, MuseScoreStyleType(styleName->currentIndex()));
+                                                },
+                                          [&]() { // update function
+                                                styleName->setCurrentIndex(int(preferences.globalStyle()));
+                                                }),
+      };
+
+      audioRelatedWidgets = vector<PreferenceItem*>{
+                  new BoolPreferenceItem(PREF_IO_ALSA_USEALSAAUDIO, alsaDriver, [](){;}),
```

Comment:
is it possible to put a default value for this third argument here?
To avoid having to type this weird character sequence every time.

Reply:
Well, the default value is `nullptr` which tells the *PreferenceItem to use the default `save` and `update` methods. Using an empty lambda means "don't do anything".

Reply:
A macro could be useful in this case. If an appropriate name was chosen.

Reply:
> Well, the default value is `nullptr` which tells the *PreferenceItem to use the default `save` and `update` methods. Using an empty lambda means "do nothing".

Oh that's why... Well it's not that bad anyway. Plus, it shouldn't happen often, so don't bother.

Reply:
> A macro could be useful in this case. If an appropriate name was chosen.

It’s a good idea to name this expression.
You can do so

```
      const auto DO_NOTHING = [](){};

      audioRelatedWidgets = std::vector<PreferenceItem*>{
                  new BoolPreferenceItem(PREF_IO_ALSA_USEALSAAUDIO, alsaDriver, DO_NOTHING),
      ...
```